components/constellation/constellation.rs

```diff
@@ -1823,7 +1836,14 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         if webdriver_reset {
             self.webdriver.load_channel = None;
         }
-        self.compositor_proxy.send(ToCompositorMsg::LoadComplete);
+        let pipeline_is_top_level_pipeline = self.browsing_contexts
+            .get(&BrowsingContextId::from(top_level_browsing_context_id))
+            .map(|ctx| ctx.pipeline_id == pipeline_id)
+            .unwrap_or(false);
+        if pipeline_is_top_level_pipeline {
+            // Notify embedder top level document finished loading.
+            self.compositor_proxy.send(ToCompositorMsg::LoadComplete(top_level_browsing_context_id));
+        }
```

Comment:
`handle_load_complete_msg` and `handle_load_start_msg` use 2 different ways of checking if the pipeline is the top level pipeline. load_start checks if there's a parent_info, load_complete checks if the browsing_context's pipeline is the same as the pipeline argument.

These 2 ways are both valid, right?

If so, which would is the most sensible?